### PR TITLE
Adding the edit functionality in user profile

### PR DESCRIPTION
**Pull request by Snehil112**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -84,6 +84,53 @@ def register_user(conn, username, first_name, last_name, email, password):
...
+            cursor.execute(update_user_query, query_params)
+
+            conn.commit()
+            cursor.close()
+            return True
+        except (Exception, psycopg2.Error) as error:
+            print("Error while updating the user:", error)
+        finally:
+            conn.close()
+    return False
```
Comment (TaariqMansurie) [inline app.py:132]: Great work snehil. It's looking perfect!

File: static/index.js [hunk 2]
```diff
@@ -1,57 +1,71 @@
...
-  modal.show();
-});
-
-document.getElementById('loginLink').addEventListener('click', function () {
-  // Open the modal
-  var modal = new bootstrap.Modal(document.getElementById('modalLoginForm'));
-  modal.show();
-});
-
-document.addEventListener('DOMContentLoaded', function () {
```
Comment (TaariqMansurie) [inline static/index.js:13]: It's always good to remove the unnecessary part of code. seems good

**Review by TaariqMansurie — COMMENTED** [review]
From my side, it's good to go 👍

File: app.py [hunk 3]
```diff
@@ -84,6 +84,53 @@ def register_user(conn, username, first_name, last_name, email, password):
...
+def user_exists(conn, username):
+    try:
+        cursor = conn.cursor()
+        check_user_query = "SELECT * FROM se_project.users WHERE username = %s"
+        cursor.execute(check_user_query, (username,))
+        existing_user = cursor.fetchone()
+        return existing_user is not None
+    except (Exception, psycopg2.Error) as error:
+        print("Error while checking if user exists:", error)
+        return False
```
Comment (KattaManisha) [inline app.py:96]: It's good that you are checking whether the username already exists or not.

**Review by KattaManisha — APPROVED** [review]
This looks good so far 👍🏻 . It would be great if we inculcate this functionality to the signup form.

**Review by TaariqMansurie — APPROVED** [review]
From my side, it's good to go. 👍

**Pull request merged**